### DISCUSSION TITLE
verifyIdToken returns Firebase ID token has expired when IDLE, setAuthCookies not refreshing the token, works on page refresh

### DISCUSSION THREAD

**Discussion by Felixaverlant** [gladly-team/next-firebase-auth#605] [Q&A] [answered]
Hi there,

**Problem**
When IDLE on the page for 1h, the token expires and any action on the page raise the following error.
`FirebaseAuthError: Firebase ID token has expired. Get a fresh ID token from your client app and try again (auth/id-token-expired). See https://firebase.google.com/docs/auth/admin/verify-id-tokens for details on how to retrieve an ID token.`

If I call setAuthCookies(req, res), I have the following error:
`FirebaseAuthError: uid argument must be a non-empty string uid.`

If I refresh the page, the token is updated and everything works fine.

Note : both `onVerifyTokenError` & `onTokenRefreshError` is not called.

**Comment by kmjennison** [comment, marked as answer]
This shouldn't happen: we'll update the token when the Firebase JS SDK refreshes it ([source](https://github.com/gladly-team/next-firebase-auth/blob/v1.x/src/useFirebaseUser.js#L157)).

Have you confirmed this also happens on the [demo app](https://github.com/gladly-team/next-firebase-auth#demo)? If it doesn't, please double-check that you're using `useAuthUser` as intended.

**Comment by Felixaverlant** [comment]
Hi Kevin,

Thank you for your fast reply! 
I do have a wrapper around useAuthUser to avoid hooks boilerplates, I'll look into it.

Regarding my issue, I guess I'll first try to upgrade to `1.0.0-canary.16` to get more logs as I found out this issue : [#351](https://github.com/gladly-team/next-firebase-auth/issues/531).
I am currently using `^1.0.0-canary.9` in production but since getFirebaseAdmin() is not available anymore, it will take me some time to fix the breaking changes.

NB: With the demo app on `1.x`, I'm able to connect, I do am signed in and access examples requiring the ID token but have these backend logs : 
```
next-firebase-auth: [setAuthCookies] Attempting to set auth cookies.
next-firebase-auth: [verifyIdToken] Error verifying the ID token: auth/argument-error. The user will be unauthenticated.
next-firebase-auth: [setAuthCookies] Failed to verify the ID token. Cannot authenticate the user or get a refresh token.
next-firebase-auth: [setAuthCookies] Set auth cookies. The user is not authenticated.
next-firebase-auth: [verifyIdToken] Error verifying the ID token: auth/argument-error. The user will be unauthenticated.
```

**Reply by kmjennison** [reply]
Sounds good. Regarding `auth/argument-error`, check out this FAQ for debugging:
https://github.com/gladly-team/next-firebase-auth#i-get-an-authargument-error-with-message-firebase-id-token-has-invalid-signature

**Reply by Felixaverlant** [reply]
Thanks I'll check it out!

**Comment by GeronimoJohn** [comment]
Heythere,

Hope everyone's good!

Just wanted to check in on that issue with tokens expiring after an hour and not getting refreshed. Has it been resolved? Currently, our app's still grappling with this problem.

Right now, we're rocking an older version, next-firebase-auth: 1.0.0-canary.16. A bit of an older version. But we're gearing up to make a big leap to 1.0.0-beta.0.

I'm kinda wondering – did that fix (if it's out there) make its way into versions beyond 1.0.0-canary.16? Can I just get the latest version and call it a day? Would love to know 😃

**Reply by Felixaverlant** [reply]
I wrote a wrapper function that I refresh the token on every ajax calls

**Reply by GeronimoJohn** [reply]
Okay that makes sense. Thanks Felix!
